### PR TITLE
Devise mailgun

### PR DESCRIPTION
**Pull request by jhiatt**
https://trello.com/c/2WeUO8Fn/137-use-mailgun-on-all-devise-emails

will need to update for IMTI domain when we add it to mailgun.

Will need thorough testing of all email operations before pushing to heroku.

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -36,11 +36,19 @@
...
     api_key: ENV["MAILGUN_API_KEY"] ,
     domain: 'sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org',
   }
+  config.action_mailer.smtp_settings = {
+    :authentication => :plain,
+    :address => "smtp.mailgun.org",
+    :port => 587,
+    :domain => "sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org",
+    :user_name => "postmaster@MYDOMAIN.mailgun.org",
+    :password => "2%oiE3!QR9jB"
```
Comment (trevordjones) [inline config/environments/development.rb]: Hey - so remember we do not put passwords or tokens in github. We do ENV variables. Other people can see this and use our mailgun info now. I'll go ahead and change and get you the updated credentials.

File: config/environments/development.rb [hunk 2]
```diff
@@ -36,11 +36,19 @@
...
   config.action_mailer.mailgun_settings = {
     api_key: ENV["MAILGUN_API_KEY"] ,
     domain: 'sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org',
   }
+  config.action_mailer.smtp_settings = {
+    :authentication => :plain,
+    :address => "smtp.mailgun.org",
+    :port => 587,
+    :domain => "sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org",
+    :user_name => "postmaster@MYDOMAIN.mailgun.org",
```
Comment (trevordjones) [inline config/environments/development.rb]: This should be `postmaster@sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org`

Comment (jhiatt) [reply config/environments/development.rb]: fixed

File: config/environments/development.rb (hunk 1 above)
Comment (jhiatt) [reply config/environments/development.rb]: fixed

File: config/environments/development.rb [hunk 3]
```diff
@@ -36,11 +36,19 @@
...
     api_key: ENV["MAILGUN_API_KEY"] ,
     domain: 'sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org',
   }
+  config.action_mailer.smtp_settings = {
+    :authentication => :plain,
+    :address => "smtp.mailgun.org",
+    :port => 587,
+    :domain => "sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org",
+    :user_name => "postmaster@sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org",
+    :password => "MAILGUN_PASSWORD"
```
Comment (trevordjones) [inline config/environments/development.rb]: `ENV['MAILGUN_PASSWORD']`

File: config/environments/production.rb [hunk 4]
```diff
@@ -66,6 +66,20 @@
...
+    api_key: ENV["MAILGUN_API_KEY"] ,
+    domain: 'sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org',
+  }
+  config.action_mailer.smtp_settings = {
+    :authentication => :plain,
+    :address => "smtp.mailgun.org",
+    :port => 587,
+    :domain => "sandbox4f9920610a894b81b82f6bc37e90f1a0.mailgun.org",
+    :user_name => "postmaster@MYDOMAIN.mailgun.org",
+    :password => "2%oiE3!QR9jB"
```
Comment (trevordjones) [inline config/environments/production.rb]: Don't forget this one

**Pull request merged**